src/components/invoices/InvoiceDetails/InvoiceDetails.js

```diff
@@ -1,149 +1,141 @@
+                        <div className='container-flex text-separate'>
+                            <h3>Razem Netto</h3>
+                            <div>{poRabacie}</div>
+                        </div>
+                        <div className='container-flex text-separate'>
+                            <h3>VAT</h3>
+                            <div>{Number.parseFloat(poRabacie - price).toFixed(2)}</div>
+                        </div>
+                        <div className='container-flex text-separate'>
+                            <h3>Razem Brutto</h3>
+                            <div className='txt-highlited'>{Number.parseFloat(price).toFixed(2)} PLN</div>
+                        </div>
+                        <Button color="secondary" size="full"  >
+                            <div className="invoiceDetails-btn">
+                                <div>
+                                    <Print width={16} height={16} />
+                                    <p>Wydrukuj / zapisz</p>
+                                </div>
+                                <Set width={16} height={16} />
                             </div>
-                            <div className='container-flex text-separate'>
-                                <h3>Razem Brutto</h3>
-                                <div className='txt-highlited'>{Number.parseFloat(price).toFixed(2)} PLN</div>
+                        </Button>
+                        <Button color="neutral" size='full'>
+                            <div className='invoiceDetails-btn'>
+                                <Mail width={16} height={16} />
+                                <p> Wyślij wiadomość</p>
                             </div>
-                            <Button color="secondary" size="full"  >
-                                <div className="invoiceDetails-btn">
-                                    <div>
-                                        <Print width={16} height={16} />
-                                        <p>Wydrukuj / zapisz</p>
-                                    </div>
-                                    <Set width={16} height={16} />
-                                </div>
-                            </Button>
-                            <Button color="neutral" size='full'>
-                                <div className='invoiceDetails-btn'>
-                                    <Mail width={16} height={16} />
-                                    <p> Wyślij wiadomość</p>
-                                </div>
-                            </Button>
-                        </div>
+                        </Button>
                     </div>
-                </Fragment>
-            </div>
-        )
-    }
+                </div>
+            </Fragment>
+        </div>
+    )
 }
-export default InvoiceDetails
+
+const mapStateToProps = state => ({
+    invoices: state.invoices,
+    contractors: state.contractors
+})
```

Comment:
I will suggest you different solution - select your invoice inside mapStateToProps

```
const mapStateToProps = (state, props) => {
  const invoiceId = props.match.params.id;
  const invoice =
    (state.invoices.invoicesList &&
      state.invoices.invoicesList.find(
        ({ id }) => invoiceId === id.toString()
      )) ||
    null;
  const contractor = invoice
    ? state.contractors.contractorsList.find(
        ({ companyName }) => companyName === invoice.contractor
      )
    : null;

  return {
    invoice,
    contractor,
  };
};
```
you can extract from this code 2 selectors - invoiceSelector and contractorSelector